docs/samples/performance-testing/grpc.md

```diff
@@ -0,0 +1,107 @@
+
+![Iter8 gRPC performanc test](grpc.png)
+
+***
+
+> Performance testing of KServe inference services using HTTP is described [here](README.md). Canary testing using Prometheus metrics is described [here](../canary-testing/README.md). This tutorial focuses on performance testing of KServe inference services with gRPC endpoints. The main steps in this tutorial are:
+> 1. [Deploy an InferenceService](#deploy-an-inferenceservice)
+> 2. [Launch an Iter8 experiment](#launch-an-iter8-experiment)
+> 3. [View experiment report](#view-experiment-report)
+
+***
+
+## Deploy an InferenceService
+
+Create an InferenceService that exposes an gRPC port. The following serves the SciKit [irisv2 model](https://kserve.github.io/website/0.10/modelserving/v1beta1/sklearn/v2/#deploy-with-inferenceservice):
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: "serving.kserve.io/v1beta1"
+kind: "InferenceService"
+metadata:
+  name: "sklearn-irisv2"
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      runtime: kserve-mlserver
+      protocolVersion: v2
+      storageUri: "gs://seldon-models/sklearn/mms/lr_model"
+      ports:
+      - containerPort: 9000
+        name: h2c
+        protocol: TCP
+EOF
+```
+
+Verify that your inference service is ready.
+
+```shell
+kubectl wait --for=condition=Ready --timeout=600s isvc/sklearn-irisv2
+```
+
+## Install Iter8 CLI
+Install the Iter8 CLI using `brew` as follows. You can also install using pre-built binaries as described [here](https://iter8.tools/0.13/getting-started/install/).
+
+```shell
+brew tap iter8-tools/iter8
+brew install iter8@0.13
+```
+
+## Launch an Iter8 experiment
+Iter8 introduces the notion of an *experiment* that makes it easy to verify that your inference service is ready, generate load for the inference, collect latency and error-related metrics, and assess SLOs for performance validation. Launch the Iter8 experiment inside the Kubernetes cluster.
+
+```shell
+iter8 k launch \
+--set "tasks={ready,grpc,assess}" \
+--set ready.isvc=sklearn-irisv2 \
+--set ready.timeout=600s \
+--set grpc.protoURL=https://raw.githubusercontent.com/kserve/kserve/master/docs/predict-api/v2/grpc_predict_v2.proto \
+--set grpc.host=sklearn-irisv2-predictor-default.default.svc.cluster.local:80 \
+--set grpc.call=inference.GRPCInferenceService.ModelInfer \
+--set grpc.dataURL=https://gist.githubusercontent.com/kalantar/6e9eaa03cad8f4e86b20eeb712efef45/raw/56496ed5fa9078b8c9cdad590d275ab93beaaee4/sklearn-irisv2-input-grpc.json \
+--set grpc.warmupNumRequests=10 \
+--set assess.SLOs.upper.grpc/latency/mean=500 \
+--set assess.SLOs.upper.grpc/latency/p90=900 \
+--set assess.SLOs.upper.grpc/error-count=0 \
+--set runner=job
```

Comment:
Does this launch a kubernetes job to generate the request load?

Reply:
Yes.